### PR TITLE
SITES-33687 - URL with Query + Fragment Reordered by Teaser Core Component

### PR DESCRIPTION
**Pull request by vladbailescu**
* Updated fragment processing logic
* Added test for edge case

### CODE REVIEW

**Review by spopescu — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkUtil.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public static String decode(final String url) throws UnsupportedEncodingExceptio
      */
     public static String escape(final String path, final String queryString, final String fragment) {
         boolean pathContainsFragment = false;
-        if (StringUtils.contains(path, fragment)) {
+        if (StringUtils.contains(path, "#" + fragment)) {
```
Comment (RaduADumitru) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkUtil.java:118]: Initially thought the issue could still happen in theory if the # character were to be used in other ways in the url, but I checked and it looks like it is a reserved character based on RFC Specification for URLs, so this shouldn't happen - [see here](https://stackoverflow.com/questions/1856785/characters-allowed-in-a-url)

**Review by RaduADumitru — APPROVED** [review]
